### PR TITLE
[TOOL] Uniform sampling

### PR DESCRIPTION
**Pull request by frozar**
**Warning**: this PR is based onto #2347

Allow the `pcl_uniform_sampling` tool to deal with several formats (PCD, PLY and VTK) as input or output point cloud.

### CODE REVIEW

File: tools/uniform_sampling.cpp [hunk 1]
```diff
@@ -49,13 +53,20 @@ using namespace pcl::console;
 
 double default_radius = 0.01;
 
-Eigen::Vector4f    translation;
-Eigen::Quaternionf orientation;
+/** \brief File type for saving and loading files. */
+typedef enum FileType
```
Comment (taketwo) [inline tools/uniform_sampling.cpp]: In C++ you don't need to `typedef` enumerations. You can safely remove `typedef` and the second `FileType`.

File: tools/uniform_sampling.cpp [hunk 2]
```diff
@@ -96,22 +108,38 @@ compute (const pcl::PCLPointCloud2::ConstPtr &input, pcl::PCLPointCloud2 &output
...
+  print_value ("%d", output_.size()); print_info (" points]\n");
 
   // Convert data back
   toPCLPointCloud2 (output_, output);
 }
 
 void
-saveCloud (const string &filename, const pcl::PCLPointCloud2 &output)
+saveCloud (const string &filename, const pcl::PCLPointCloud2 &output,
+           const FileType &output_file_type)
```
Comment (taketwo) [inline tools/uniform_sampling.cpp]: `enum` is an integral type, so better passed by value.

**Comment by frozar** [comment]
You're right, the `enum FileType` is not strictly necessary.  

The logic I had in mind when I do it is that I don't want to launch any computation without a minimal checking of the argument:
 - if the input file at least exists,
 - if the ouput format at least is supported.

So this is done in the beginning of the main function and will save time with a try/error approach around this tool.

The `enum` is just there because I feel like it makes it clearer. That's my two cents.

File: tools/uniform_sampling.cpp [hunk 3]
```diff
@@ -130,10 +158,47 @@ main (int argc, char** argv)
...
   // Parse the command line arguments for .pcd files
   vector<int> p_file_indices;
-  p_file_indices = parse_file_extension_argument (argc, argv, ".pcd");
+  vector<std::string> extension;
+  extension.push_back (".pcd");
+  extension.push_back (".ply");
+  extension.push_back (".vtk");
+  p_file_indices = parse_file_extension_argument (argc, argv, extension);
+
   if (p_file_indices.size () != 2)
```
Comment (taketwo) [inline tools/uniform_sampling.cpp]: > if the ouput format at least is supported.

This line effectively checks that. If the format of the output file is not supported, then we won't have two file names parsed.

Comment (frozar) [reply tools/uniform_sampling.cpp]: You're right. So I'll make a proposition soon.

File: tools/uniform_sampling.cpp [hunk 4]
```diff
@@ -130,13 +150,29 @@ main (int argc, char** argv)
...
-    print_error ("Need one input PCD file and one output PCD file to continue.\n");
+    print_error ("Need one input file and one output file to continue.\n");
     return (-1);
   }
 
+  // Check if the input point cloud exists
+  std::string input_filename = argv[p_file_indices[0]];
+  ifstream ifile (input_filename.c_str ());
+  if (!ifile) {
+    print_error ("Cannot found input file name (%s).\n", input_filename.c_str ());
```
Comment (taketwo) [inline tools/uniform_sampling.cpp]: We can move this to line 75, where `pcl::io::load` returns read status. Otherwise we are opening input file two times.

**Pull request merged**